Unix/pal/host.c

```diff
@@ -0,0 +1,90 @@
+/*
+**==============================================================================
+**
+** Copyright (c) Microsoft Corporation. All rights reserved. See file LICENSE
+** for license information.
+**
+**==============================================================================
+*/
+
+#include <pal/strings.h>
+#include <ctype.h>
+
+#if defined(PAL_HAVE_POSIX)
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <unistd.h>
+
+#endif
+
+
+/*
+**==============================================================================
+**
+**  Host name (POSIX)
+**
+**
+**==============================================================================
+*/
+
+#if defined(PAL_HAVE_POSIX)
+
+/*
+ *  char* getFullyQualifiedDomainName()
+ *
+ *  returns a new copy of he primary full qualified domain name as utf8/ascii. 
+ *
+ */ 
+
+const char *
+getFullyQualifiedDomainName()
+
+{
+    struct addrinfo hints, *info;
+    int gai_result;
+
+    char hostname[1024];
+    hostname[1023] = '\0';
+
+    gethostname(hostname, 1023);
+
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_UNSPEC;    /*either IPV4 or IPV6 */
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_CANONNAME;
+
+    if ((gai_result = getaddrinfo(hostname, NULL, &hints, &info)) != 0)
```

Comment:
A lot of this is protocol stuff that I'm not familiar with, so I'll leave that to Paul.

Here, info appears to be something that should be freed via `freeaddrinfo`, is this done?